src/app/shared/truncatable/truncatable-part/truncatable-part.component.ts

```diff
@@ -67,10 +95,67 @@ export class TruncatablePartComponent implements OnInit, OnDestroy {
     this.sub = this.service.isCollapsed(this.id).subscribe((collapsed: boolean) => {
       if (collapsed) {
         this.lines = this.minLines.toString();
+        this.expand = false;
       } else {
         this.lines = this.maxLines < 0 ? 'none' : this.maxLines.toString();
+        this.expand = true;
+      }
+    });
+  }
+
+  ngAfterContentChecked() {
+    if (this.isBrowser) {
+      if (this.observer && this.observedContent) {
+        this.toUnobserve();
+      }
+      this.toObserve();
+    }
+  }
+
+  /**
+   * Function to get data to be observed
+   */
+  toObserve() {
+    this.observedContent = this.document.querySelectorAll('.content');
+    this.observer = new (this._window.nativeWindow as any).ResizeObserver((entries) => {
```

Comment:
This method gets called once for every truncatable component on the page, but it attaches this observer to _all_ truncatable components every time. So the size calculations happen way too often. I'd try to attach it only to the content div in the current component.

It's probably also a good idea to use `hostWindowService.widthCategory` instead of an observer directly on the window size. Because the dspace theme uses containers, the dimensions of a truncatable component only change when we go from one withcategory to another, not every time the width of the window changes. The exception is the XS category, below that you do need to respond to any resize event.

And finally since this is an expensive calculation it's a good idea to use a debounce operator. That way if the user quickly resizes their screen by a bunch at once (multiple categories) you can wait until it settles instead of recalculating every step in between